src/Manipulators/Size.php

```diff
@@ -271,102 +270,93 @@ public function runResize(Image $image, $fit, $width, $height)
     /**
      * Perform contain resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runContainResize(Image $image, $width, $height)
+    public function runContainResize(ImageInterface $image, $width, $height): ImageInterface
     {
-        return $image->resize($width, $height, function ($constraint) {
-            $constraint->aspectRatio();
-        });
+        return $image->resize($width, $height);
```

Comment:
Should this be `return $image->scale($width, $height);`?  In v3 `Image::resize()` stretches the image and the alternative to preserve the aspect ratio is `Image::scale()` as the callback param of `resize()` is gone which allowed specifying the aspect ratio constraint.

Reply:
You are right. Using `?h=600&w=500&size=contain` results in a square image. Using `scale()` know delivers the same result as before.

![grafik](https://github.com/thephpleague/glide/assets/2162994/4948174b-858b-40a0-9ce2-b0a1a8a4f44b)